File: api/health.js
module.exports = async (req, res) => {
    // Set CORS headers
    res.setHeader('Access-Control-Allow-Origin', '*');
    res.setHeader('Access-Control-Allow-Methods', 'GET, OPTIONS');
    res.setHeader('Access-Control-Allow-Headers', 'Content-Type');

    // Handle preflight requests
    if (req.method === 'OPTIONS') {
        return res.status(200).end();
    }

    if (req.method === 'GET') {
        return res.status(200).json({
            status: 'healthy',
            service: 'Telegram ID Analyzer Bot',
            timestamp: new Date().toISOString(),
            version: '1.0.0',
            runtime: 'Node.js on Vercel'
        });
    }

    return res.status(405).json({ error: 'Method not allowed' });
};

File: api/health.rs
use vercel_runtime::{run, Body, Error, Request, Response};
use serde_json::json;

#[tokio::main]
async fn main() -> Result<(), Error> {
    run(handler).await
}

pub async fn handler(_req: Request) -> Result<Response<Body>, Error> {
    let health_info = json!({
        "status": "healthy",
        "service": "Telegram ID Analyzer Bot",
        "timestamp": chrono::Utc::now().to_rfc3339(),
        "version": "1.0.0",
        "runtime": "Rust on Vercel"
    });

    Ok(Response::builder()
        .status(200)
        .header("Content-Type", "application/json")
        .header("Access-Control-Allow-Origin", "*")
        .body(health_info.to_string().into())?)
}